### PR TITLE
Fix WiFiClientRxBuffer::fillBuffer() error, likely solves issue #2212

### PR DESCRIPTION
**Pull request by Jeroen88**
Fix WiFiClientRxBuffer::fillBuffer() error in handing a negative res from the call to recv()

In https://github.com/espressif/arduino-esp32/blob/229d9b7366deee0d8a90fb69c19119f7287c9d1d/libraries/WiFi/src/WiFiClient.cpp#L70 if res is negative and errno equals EWOULDBLOCK then the if() condition is false causing _fill to be decremented. 
The condition res < 0 && errno == EWOULDBLOCK simply means that the socket has no data available (yet), so _fill should not change and, since nothing is read, the return value of fillBuffer should be zero.

This most likely solves the complete #2212 issue, at least it solves the minimum sketch reproducing the error as posted in that issue by @lbernstone. This could also be the root cause of other issues, since on slow connections it is likely that the socket not always has data available.

I also added an extra test on https://github.com/espressif/arduino-esp32/blob/229d9b7366deee0d8a90fb69c19119f7287c9d1d/libraries/WiFi/src/WiFiClient.cpp#L60 by setting _failed to true if the malloc() fails. I am not sure and did not check if the _failed condition is checked in the other functions, but it should not do any harm. Moreover, this did not cause the error.

Happy New Year to everyone!

### CODE REVIEW

File: libraries/WiFi/src/WiFiClient.cpp [hunk 1]
```diff
@@ -58,6 +58,10 @@ class WiFiClientRxBuffer {
         {
             if(!_buffer){
                 _buffer = (uint8_t *)malloc(_size);
+                if(!_buffer) {
```
Comment (copercini) [inline libraries/WiFi/src/WiFiClient.cpp]: Please add an error log like `log_e("Not enough memory to allocate buffer");`

this helps a lot on debug =)

**Comment by lbernstone** [comment]
I'm putting a different fix in for this that seems closer to the root.  Put em both out there and see what shakes out, I figure...

**Comment by Jeroen88** [comment]
@lbernstone, @nikuz confirmed that my PR fixes the error. Also, the if() in WiFiClientRxBuffer::fillBuffer() is obviously wrong...

**Comment by lbernstone** [comment]
Verified.  Looks like you included some BLE stuff in your commit, tho.

**Comment by Jeroen88** [comment]
@lbernstone You are right, I always pull master before creating a branch. So I was surpised to have some BLE stuff in my master

**Comment by me-no-dev** [comment]
I can not merge this with the BLE stuff in there ;)

**Comment by Jeroen88** [comment]
@me-no-dev can you help me get it out? I did a git pull upstream master and then it got in...????

**Comment by me-no-dev** [comment]
delete the BLE folder then call `git submodule update --init --recursive` in the root arduino folder

**Comment by Jeroen88** [comment]
Replaced by #2263

**Pull request closed without merging**